client/jetpack-cloud/components/user-feedback-modal-form/index.tsx

```diff
@@ -52,19 +55,37 @@ export default function UserFeedbackModalForm( { show, onClose }: Props ) {
 		setFeedback( event.currentTarget.value );
 	}, [] );
 
-	const onRatingChange = useCallback( ( rating: number ) => {
-		setRating( rating );
-	}, [] );
+	const onRatingChange = useCallback(
+		( rating: number ) => {
+			dispatch( recordTracksEvent( 'calypso_jetpack_user_feedback_form_rating_click' ) );
+			setRating( rating );
+		},
+		[ dispatch ]
+	);
 
 	const hasCompletedForm = !! feedback && !! rating;
 
 	const onSubmit = useCallback( () => {
 		if ( ! hasCompletedForm ) {
 			return;
 		}
+
+		dispatch(
+			recordTracksEvent( 'calypso_jetpack_user_feedback_form_submit', {
+				rating,
+				feedback,
+			} )
+		);
+
 		const sourceUrl = `${ window.location.origin }${ window.location.pathname }`;
 		submitFeedback( { feedback, rating, source_url: sourceUrl } );
-	}, [ feedback, hasCompletedForm, rating, submitFeedback ] );
+	}, [ dispatch, feedback, hasCompletedForm, rating, submitFeedback ] );
+
+	useEffect( () => {
+		if ( show ) {
+			dispatch( recordTracksEvent( 'calypso_jetpack_user_feedback_form_open' ) );
```

Comment:
We are already tracking this here: `calypso_jetpack_sidebar_share_product_feedback_click`. I don't think this is needed.

Reply:
We have an option to open the form via the link, so this one ensures we cover all scenarios. I'm not really sure if `calypso_jetpack_sidebar_share_product_feedback_click` is necessary but I feel like it will be useful for funnel analysis.